lms/djangoapps/completion/models.py

```diff
@@ -147,3 +147,38 @@ def __unicode__(self):
             block_key=self.block_key,
             completion=self.completion,
         )
+
+
+class AggregateCompletion(TimeStampedModel):
+    """
+
+    Aggregators are blocks that contain other blocks, are not themselves completable,
+    and are considered complete when all descendant blocks are complete.
+
+    """
+    id = BigAutoField(primary_key=True)  # pylint: disable=invalid-name
+    user = models.ForeignKey(User)
+    course_key = CourseKeyField(max_length=255)
+    aggregation_name = models.CharField(max_length=255)
+    block_key = UsageKeyField(max_length=255)
+    earned = models.FloatField()
+    possible = models.FloatField()
+    percent = models.FloatField(validators=[validate_percent])
+
```

Comment:
This should have a manager like `BlockCompletionManager` above, which provides a convenience method that will set fields like `percent` automatically, and which can validate the other fields and provide the insert_or_update functionality needed to update records.

Reply:
I have pushed up another version to let me know if this is in the realm that we're looking to take it that includes the manager and submit_completion method.